### ISSUE TITLE
why need transpose in adjacency matrix normalization? 

### ISSUE THREAD

**Issue by yanlirock** [tkipf/gcn#142]
in the function normalize_adj
the output is:
"adj.dot(d_mat_inv_sqrt).transpose().dot(d_mat_inv_sqrt).tocoo()"

why not using 
"d_mat_inv_sqrt.dot(adj).dot(d_mat_inv_sqrt) "
follow by the original equation in paper?

and the results of above two are different

**Comment by MeetUo** [comment]
I also have this confusion.
But I find  the adj matrix is the symmetric matrix and the d_mat_inv_sqrt is the diagonal matrix .
So 
"adj.dot(d_mat_inv_sqrt).transpose().dot(d_mat_inv_sqrt)"
actually equal
"d_mat_inv_sqrt.dot(adj).dot(d_mat_inv_sqrt) "
But I also don't know why need transpose, hope the author can answer our doubts.

**Comment by leeswan** [comment]
transpose for construct symmetric normalized adj matrix in tensorflow version, while in pytorch version, the method of normalization is different, features and adj take the same way

**Comment by kentfly** [comment]
(A*B)T = BT * AT
so..  "adj.dot(d_mat_inv_sqrt).transpose().dot(d_mat_inv_sqrt)"
actually **not** equal to 
"d_mat_inv_sqrt.dot(adj).dot(d_mat_inv_sqrt) "

I hope the author can explain this problem which is so important for the GCN...thx...

> I also have this confusion.
> But I find the adj matrix is the symmetric matrix and the d_mat_inv_sqrt is the diagonal matrix .
> So
> "adj.dot(d_mat_inv_sqrt).transpose().dot(d_mat_inv_sqrt)"
> actually equal
> "d_mat_inv_sqrt.dot(adj).dot(d_mat_inv_sqrt) "
> But I also don't know why need transpose, hope the author can answer our doubts.

**Comment by tkipf** [comment]
If `adj` is symmetric (which we assume) then both are identical.

**Comment by kentfly** [comment]
i get it!!
thx a looooot....:))
